libraries/bootstrap.php

```diff
@@ -41,6 +41,21 @@ class_exists('\\Joomla\\CMS\\Autoload\\ClassLoader');
 // Decorate Composer autoloader
 spl_autoload_register([new \Joomla\CMS\Autoload\ClassLoader($loader), 'loadClass'], true, true);
 
+// Check if load of the environment variables is enabled.
+if (is_file(JPATH_ROOT . '/.env') || is_file(JPATH_ROOT . '/.env.dev')) {
+    /**
+     * Some PHP configurations set variables_order="GPCS" not "EGPCS", in which case $_ENV is NOT populated.
+     * Detect if the $_ENV  was empty and handle it by explicitly populating through getenv() for CLI,
+     * or filter variables from  $_SERVER prefixed with JOOMLA_ for web requests.
+     * To make it work the same for CLI and WEB set variables_order="EGPCS" in php.ini.
+     */
+    if (!$_ENV) {
+        $_ENV = PHP_SAPI === 'cli' ? getenv() : array_filter($_SERVER, fn ($k) => str_starts_with($k, 'JOOMLA_'), ARRAY_FILTER_USE_KEY);
+    }
+
+    Dotenv\Dotenv::createImmutable(JPATH_ROOT, ['.env', '.env.dev'], false)->safeLoad();
```

Comment:
I think we should either drop the `.env.dev` file or introduce a special environment variable e.g. `JOOMLA_ENV` (`prod`, `dev`, `test` and so on):

```suggestion
    Dotenv\Dotenv::createImmutable(JPATH_ROOT, ['.env', '.env.'.($_ENV['JOOMLA_ENV'] ?? 'prod')], false)->safeLoad();
```

Reply:
I do not really see a need in it. Keep it simple for now

Reply:
it is now like that